### DISCUSSION TITLE
argocd getting error connecting to Azure devops repo

### DISCUSSION THREAD

**Discussion by dlcctg** [argoproj/argo-cd#9691] [Q&A]
When trying to connect to Azure Devops repo (using the gui under settings repos with the following git@ssh.dev.azure.com:v3) we are seeing the message - unable to load data, failed to add RSA key hosts to the list of known hosts you are not authorized to connect to the permissions.  
We are also see this app conditions error - 
rpc error: code = Unknown desc = unknown error: remote: You are not authorized to access this collection.  This was working fine.  The only change is we had an employee leave the company who helped manage the environment.  His username was removed but there are other admins still working on the platform under the argoadmin group in Openshift(kubernetes).

Thanks for the help!

**Comment by 34fathombelow** [comment]
By any chance have you recently upgraded to Argo CD 2.4?  If so, in 2.4 the base image was upgraded to Ubuntu 22.04.  SHA1 ssh signature algorithms are no longer supported by default.   This is something that needs to be upgraded on [ssh.dev.azure.com](ssh.dev.azure.com) side. 

For the workaround and more explanation check `https://argo-cd.readthedocs.io/en/latest/operator-manual/upgrading/2.3-2.4/#support-for-private-repo-ssh-keys-using-the-sha-1-signature-hash-algorithm-is-removed` if you need a detailed example of the workaround check #9640

**Reply by dlcctg** [reply]
We are running ARGOCD 2.3.4
